pandas/tools/tile.py

```diff
@@ -81,6 +83,17 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
     array([1, 1, 1, 1, 1], dtype=int64)
     """
     # NOTE: this binning code is changed a bit from histogram for var(x) == 0
+    # for handling the cut for datetime and timedelta objects
+
```

Comment:
Yes, we should preserve the index. So needs below logic before converting to array.

- https://github.com/aileronajay/pandas/blob/b21fa237537ff225fca955355000a7b2fc2a2bb4/pandas/tools/tile.py#L204